### DISCUSSION TITLE
Typing Help: setValue's behavior for TFieldValues ​​containing nullable values

### DISCUSSION THREAD

**Discussion by berlysia** [react-hook-form/react-hook-form#2915] [Q&A] [answered]
I want to pass a cached object in a higher context to the form.
But I can't use `reset` because I don't want to update the defaultValues.

I'm trying to use `setValue`.
I want to keep it the component does not depend on the structure of the object, use iterated keys of the object.

I got some error when some value is nullable.
How do I deal with this?

react-hook-form version: v6.8.1
TypeScript version: v4.0.2

```ts
type MyObj = {
  nullableNumber: number | null;
}
const obj: MyObj = {
  nullableNumber: 123,
};

const Comp: React.FC = () => {
  const { setValue } = useForm<MyObj>({ defaultValues: obj });

  // pass typecheck
  setValue("nullableNumber", null);

  // error TS2345: Argument of type 'null' is not assignable to parameter of type `{} | undefined'
  setValue("nullableNumber" as string, null);

  // same as above, at `obj[key]`
  Object.keys(obj).forEach(key => setValue(key, obj[key]));

  // I've found a tweak(I'm sure that this is unsafe, but enough for myself)
  Object.keys(obj).forEach(key => setValue(key as keyof MyObj, obj[key]));
};
```

**Comment by bluebill1049** [comment, marked as answer]
I think we will fix setValue to allow null

**Reply by ghost** [reply]
Is it fixed? I'm trying to set `null` using `setValue` and it sets `false` instead of `null` the field is controlled through the controller...

**Reply by bluebill1049** [reply]
can you share a codesndbox?
